kong/db/declarative/init.lua

```diff
@@ -398,6 +398,22 @@ function declarative.load_into_db(entities, meta)
 end
 
 
+local function begin_transaction(db)
+  if db.strategy == "postgres" then
+    db.connector:connect("read")
+    db.connector:query("BEGIN TRANSACTION ISOLATION LEVEL REPEATABLE READ READ ONLY;", "read")
```

Comment:
Should we check the return value here? Same for the `commit_transaction` function below.

Reply:
I made it fail hard on begin transaction.